projet/projects/tests/test_contribution.py

```diff
@@ -18,52 +18,53 @@ def setUp(self):
-        )
+        # self.accept_contributor = reverse(
+        #     "accept-contribution", args=[self.project.id, self.contribution.id]
+        # )
         self.project_contributions = reverse("contributors", args=[self.project.id])
         self.project_contributions_status = reverse(
             "pending-accepted-contributors", args=[self.project.id, "accepted"]
         )
+        self.invited = User.objects.create_user("invite@email.com", "somethingwoth")
 
     def test_owner_can_be_contributor(self):
         self.client.force_login(user=self.second_user)
-        contribute_url = reverse("contribute", args=[self.second_project.id])
-        response = self.client.post(contribute_url, format="json")
+        contribute_url = reverse("invite")
+        data = {"id": f"{self.second_project.id}", "email": f"{self.second_user.email}"}
 
-        self.assertEqual(response.status_code, 403)
+        response = self.client.post(contribute_url, data=data, format="json")
 
-    def test_contributor_can_offer_twice(self):
-        self.client.force_login(user=self.third_user)
-        contribute_url = reverse("contribute", args=[self.project.id])
-        response = self.client.post(contribute_url, format="json")
-        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.status_code, 401)
 
-    def test_manager_can_accept_contribution(self):
-        self.client.force_login(user=self.user)
-        accept_url = reverse(
-            "accept-contribution", args=[self.project.id, self.contribution.id]
+    def test_user_can_be_invited_or_contributor(self):
+        self.client.force_login(user=self.second_user)
+        data = {"id": f"{self.second_project.id}", "email": f"{self.third_user.email}"}
+        invite_url = reverse("invite")
+        response = self.client.post(invite_url, data=data, format="json")
+        self.assertEqual(response.status_code, 201)
+
+    def test_user_can_be_invited_twice(self):
+        self.client.force_login(user=self.second_user)
+        new_user = User.objects.create_user("new@email.com", "newusertotest")
+        contributor = Contributor.objects.create(
+            user=new_user, project=self.second_project
         )
+        data = {"id": f"{self.second_project.id}", "email": f"{new_user.email}"}
+        invite_url = reverse("invite")
+        response = self.client.post(invite_url, data=data)
+        self.assertEqual(response.status_code, 400)
 
-        response = self.client.patch(accept_url, data={"accepted": True}, format="json")
+    def test_user_can_accept_contribution(self):
+        self.client.force_login(user=self.third_user)
+        accept_url = reverse("accept-contribution", args=[self.project.id])
+
+        response = self.client.patch(accept_url, format="json")
         cont = Contributor.objects.get(project=self.project.id)
 
         self.assertEqual(response.status_code, 202)
-
-    def test_contributor_can_accept_self(self):
```

Comment:
Also became obsolete because the user will have to accept the invitation himself